src/snowflake/snowpark/mock/mock_analyzer.py

```diff
@@ -546,15 +545,6 @@ def do_resolve_with_resolved_children(
     ) -> MockExecutionPlan:
         if isinstance(logical_plan, MockExecutionPlan):
             return logical_plan
-
-        # Re-wrap SelectStatement into a MockSelectStatement
```

Comment:
We shouldn't have this. Let's completely separate `MockSelectStatment` and `SelectStatement`, these two should not be used together.